app/services/validator.py

```diff
@@ -10,21 +20,94 @@ def validate_classTrib(data):
         if data['infoContri']['inclusao']['infoCadastro']['clasTrib'] in [21, 22]:
             raise app_exceptions.InvalidInput("clasTrib inválido")
 
-def validate_email(data):
+
+def validate_emailCtt(data):
     email = data['infoContri']['inclusao']['infoCadastro']['contato']['email']
+    validate_email(email)
+
+
+def validate_email(email):
     regex = '^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$'
-    if not(re.search(regex,email)):
+    if not(re.search(regex, email)):
         raise app_exceptions.InvalidInput("Email inválido")
 
-def validate_cpf(data):
+
+def validate_cpfCtt(data):
     cpf = data['infoContri']['inclusao']['infoCadastro']['contato']['cpfCtt']
+    validate_cpf(cpf)
+
+
+def validate_cpf(cpf):
     if len(cpf) != 11:
-        raise app_exceptions.InvalidInput("CPF com tamanho inválido")
+        raise app_exceptions.InvalidInput(
+            "CPF com tamanho inválido")
     try:
-        for i in range(9,11):
+        for i in range(9, 11):
             value = sum((int(cpf[num]) * ((i+1) - num) for num in range(0, i)))
             digit = ((value * 10) % 11) % 10
             if str(digit) != cpf[i]:
                 raise app_exceptions.InvalidInput("CPF inválido")
     except Exception:
-        raise app_exceptions.InvalidInput("CPF inválido")
\ No newline at end of file
+        raise app_exceptions.InvalidInput("CPF inválido")
+
+
+def validate_nrInsc(data):
+    tpInscricao = data['ideContri']['tpInsc']
+    nrInsc = data['ideContri']['nrInsc']
+    indSitPessoaJuridica = data['infoContri']['inclusao']['infoCadastro']['indSitPJ']
+
+    if(tpInscricao == tpInsc.cnpjTp):
+        if(indSitPessoaJuridica == indSitPJ.incorporacao):
+            raise app_exceptions.InvalidInput(
+                "CNPJ não poderá pertencer a pessoa jurídica inapta")
+        else:
+            validate_cnpj(nrInsc)
+    else:
+        validate_cpf(nrInsc)
+
+
+def validate_cnpjEFR(data):
+    cnpj = data['infoContri']['inclusao']['infoCadastro']['infoEFR']['cnpjEFR']
+    if(cnpj):
+        validate_cnpj(cnpj)
```

Comment:
na linha 77, já não está sendo validado se o cnpj existe de maneira mais elegante? não seria melhor deletar esse metodo?

Reply:
Coloquei esse if pois o campo cnpjEFR não é obrigatório

Reply:
Aí teoricamente só vai validar se ele for preenchido